### DISCUSSION TITLE
Official Docker image support

### DISCUSSION THREAD

**Discussion by Atydin** [investbrainapp/investbrain#9] [Ideas]
Would it be possible to publish an official Docker image for this project? This would greatly simplify deployment and version management for users and maintainers. Thank you for considering this request.

**Comment by hackeresq** [comment]
Hi @Atydin updating this to a discussion since it's not an "issue". 

Agreed on simplifying deployment and version management. I'll look into this!

**Reply by Marx2** [reply]
I've managed to create GHA for building docker image:
https://github.com/Marx2/investbrain/blob/main/.github/workflows/build.yml
Image builds successfully.
Unfortunately this image doesn't start, as it does some magic, which in general should not be done:
- installing redis (should be separate container, like mysql)
- installing of a lot of dependencies - should be installed during image build, not during startup

Investbrain also lacks of standard probes, like healtcheck probe or rediness probe.
It limits usability in Kubernetes deployment.

**Reply by hackeresq** [reply]
Unfortunately, I'm not an expert with Kubernetes and unsure what it would require to run. 

As far as the "magic" - these parts of the `entrypoint.sh` should really only be running once (i.e. creating an .env file, setting up the env with app keys and readying the filesystem). If not, there's something wrong.

Agreed on Redis. Seems like that should be running in its own container. Seems like something that can be easily moved to the `docker-compose.yml` file.  I'll work on that this weekend!

Open to any PRs that would make a Kubernetes deployment possible.

**Reply by hackeresq** [reply]
Got super sucked into this... I created a branch that uses a separate Docker image for Investbrain (also a separate service for Redis).

Curious if [this](https://github.com/investbrainapp/investbrain/tree/create-docker-image) would run in a Kubernetes environment? 

As far as I can tell, this branch would not be a breaking change for current users.

**Reply by Marx2** [reply]
I'll check it as soon as possible and let you know. 
What I can also do is to invite you in into k8s homelabers community at Discord: https://discord.gg/pQukJZQq
People there can help you with kubernetes things. For example Immich developer is there. Other nice thing is you don't need to write your own helm chart, as there is universal one written by other member: https://bjw-s.github.io/helm-charts/docs/app-template/

**Reply by Marx2** [reply]
I've tried to run this image and I'm getting an error: 
`exec /bin/bash: exec format error`

**Reply by hagai-biton2** [reply]
I have sort of workaround for that,
you need to use the Dockerfile + the docker-compose.yml from the 1.0.7 tag 
 then after building the docker image you can load it from local storage like this:
  networks:
  investbrain-network:
    driver: bridge
services:
  app:
    image: investbrain-app:latest
    container_name: investbrain-app
    restart: unless-stopped
    
    ##rest of the compose

**Comment by hackeresq** [comment]
Since @Atydin's original suggestion was implemented in v1.0.8, I will be closing this discussion. 

But I do want to explore K8s compatibility more. Moving the ongoing K8s discussion over to #43.
